src/app/features/quest/quest-preview/quest-preview.service.ts

```diff
@@ -304,4 +305,54 @@ export class QuestPreviewService {
     return Number(value);
   }
 
+  getObjText(field: string | number) {
+    return this.questTemplate[`ObjectiveText${field}`];
+  }
+
+  getObjective$(field: string | number): Promise<string> {
+
+    const RequiredNpcOrGo = Number(this.questTemplate[`RequiredNpcOrGo${field}`]);
+    if (!!RequiredNpcOrGo) {
+      if (RequiredNpcOrGo > 0) {
+        return this.mysqlQueryService.getCreatureNameById(RequiredNpcOrGo);
+      }
+
+      return this.mysqlQueryService.getGameObjectNameById(Math.abs(RequiredNpcOrGo));
+    }
+
+  }
+
+  getObjectiveCount(field: string | number): string {
+    const reqNpcOrGo = this.questTemplate[`RequiredNpcOrGoCount${field}`];
+    return !!reqNpcOrGo && reqNpcOrGo > 1 ? `(${reqNpcOrGo})` : '';
+  }
+
+  isNpcOrGoObj(field: string | number): boolean {
+    return !!this.questTemplate[`RequiredNpcOrGoCount${field}`];
+    // return !!this.questTemplate[`ObjectiveText${field}`] || !!this.questTemplate[`RequiredNpcOrGo${field}`];
```

Comment:
?

Reply:
they should be equivalent, I would keep it as a comment because in future may we can understand that it is better the second return